builtin/logical/mongodb/path_config_lease.go

```diff
@@ -0,0 +1,107 @@
+package mongodb
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/hashicorp/vault/logical"
+	"github.com/hashicorp/vault/logical/framework"
+)
+
+func pathConfigLease(b *backend) *framework.Path {
+	return &framework.Path{
+		Pattern: "config/lease",
+		Fields: map[string]*framework.FieldSchema{
+			"ttl": &framework.FieldSchema{
+				Type:        framework.TypeString,
+				Description: "Default ttl for credentials.",
+			},
+
+			"max_ttl": &framework.FieldSchema{
+				Type:        framework.TypeString,
+				Description: "Maximum time a set of credentials can be valid for.",
+			},
+		},
+
+		Callbacks: map[logical.Operation]framework.OperationFunc{
+			logical.ReadOperation:   b.pathConfigLeaseRead,
+			logical.UpdateOperation: b.pathConfigLeaseWrite,
+		},
+
+		HelpSynopsis:    pathConfigLeaseHelpSyn,
+		HelpDescription: pathConfigLeaseHelpDesc,
+	}
+}
+
+func (b *backend) pathConfigLeaseWrite(
+	req *logical.Request, d *framework.FieldData) (*logical.Response, error) {
+	ttlRaw := d.Get("ttl").(string)
+	maxTTLRaw := d.Get("max_ttl").(string)
+
+	ttl, err := time.ParseDuration(ttlRaw)
+	if err != nil {
+		return logical.ErrorResponse(fmt.Sprintf(
+			"Invalid ttl: %s", err)), nil
+	}
+	maxTTL, err := time.ParseDuration(maxTTLRaw)
+	if err != nil {
+		return logical.ErrorResponse(fmt.Sprintf(
+			"Invalid max_ttl: %s", err)), nil
+	}
+
+	// Store it
+	entry, err := logical.StorageEntryJSON("config/lease", &configLease{
```

Comment:
See above comment. After changing it this entire section above can basically be removed and you can just set the TTL/MaxTTL values below with `d.Get("ttl"/"max_ttl").(int)`. They'll return 0 if not set.